### DISCUSSION TITLE
How to allow my package consumer to access one of my dependencies cli?

### DISCUSSION THREAD

**Discussion by ValorHeart** [pnpm/pnpm#6782] [Q&A]
Hello, pretty new at PNPM,
I have my package that is being consumed by other projects.
My package has a dependency that published its cli to run it.
Using regular NPM, it would work as usual, the consumer would install my package and automatically has access to my dependency CLI.
I understand that it isn't like that using PNPM, as it doesn't work :)
What is the correct way to grant access to my package consumers to access my dependency package cli?

Thanks, Adi

**Comment by KSXGitHub** [comment]
I would create a transitory bin with the following content:

```javascript
#! /usr/bin/env node
const { bin } = require('my-dependency/package.json')
const dir = path.dirname(require.resolve('my-dependency/package.json'))
const realBin = path.join(dir, bin)
require(realBin)
```
